daemon/docker_watcher.go

```diff
@@ -123,19 +124,33 @@ func (d *Daemon) processEvent(m dTypesEvents.Message) {
 }
 
 func getCiliumEndpointID(cont *dTypes.ContainerJSON, gwIP *addressing.NodeAddress) uint16 {
-	for _, contNetwork := range cont.NetworkSettings.Networks {
-		ipv6gw := net.ParseIP(contNetwork.IPv6Gateway)
-		if ipv6gw.Equal(gwIP.IPv6Address.IP()) {
-			ip, err := addressing.NewCiliumIPv6(contNetwork.GlobalIPv6Address)
-			if err == nil {
-				return ip.EndpointID()
-			}
-		}
+	if cont.NetworkSettings == nil {
+		return 0
 	}
 
+	if ciliumIP := getCiliumIPv6(cont.NetworkSettings.Networks, gwIP); ciliumIP != nil {
+		return ciliumIP.EndpointID()
+	}
 	return 0
 }
 
+func getCiliumIPv6(networks map[string]*dNetwork.EndpointSettings, gwIP *addressing.NodeAddress) *addressing.CiliumIPv6 {
```

Comment:
Docker libnetworks differentiates between multiple networks, should we check if the driver type of the network is Cilium? I'm not familiar with docker network to understand if conflicting IPs across Docker networks is a supported model.